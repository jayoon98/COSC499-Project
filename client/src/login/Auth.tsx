import React, { useEffect } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { getCachedUserCredentials } from '../services/login';
export const MAIN_SCENE = 'questionnaires';

// TODO: Pass default page as prop in case it changes
export function AuthHandler() {
  useEffect(() => {
    getCachedUserCredentials((res, _) => {
      if (res) {
        Actions.replace(MAIN_SCENE);
        return;
      }
      Actions.replace('login');
    });
  }, []);

  return <View style={styles.container}></View>;
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    display: 'flex',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    marginTop: 100,
    padding: 48,
  },
});
