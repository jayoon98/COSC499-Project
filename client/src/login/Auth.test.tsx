import React from 'react';
import renderer from 'react-test-renderer';
import { AuthHandler } from './Auth';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

jest.mock('@react-native-community/async-storage', () => ({
  AsyncStorage: '',
}));

it('renders correctly', () => {
  const tree = renderer.create(<AuthHandler />).toJSON();
  expect(tree).toMatchSnapshot();
});
