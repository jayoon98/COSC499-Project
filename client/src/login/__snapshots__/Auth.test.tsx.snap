// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "display": "flex",
      "height": "100%",
      "marginTop": 100,
      "padding": 48,
      "width": "100%",
    }
  }
/>
`;
